### ISSUE TITLE
Javascript .query() method doesn't work

### ISSUE THREAD

**Issue by bbirand** [influxdb/influxdb#2447]
I had some client-side JS code written a few weeks ago that would work normally with InfluxDB. I just checked back now, and it no longer works. I couldn't get to the bottom of the issue, but it seems that it may have been caused by an update to the JS file..

This is the code:

```javascript
influxdb = new InfluxDB({
  "host" :"104.131.XX.XX",
  "port" :"8086",
  "username" :"root",
  "password" :"root",
  "database" :"data"
  });
influxdb.query("select min(value) from a_series group by time(1d)")
    .then(function(a) { ... })
```

This creates this query:
```bash 
curl 'http://104.131.XX.XX:8086/query?u=root&p=root&q=select%20min(value)%20from%20a_series%20group%20by%20time(1d)'
```

And returns this error:
```javascript
{"results":[{"error":"database not found: "}]}
```

When I try to run the query manually from within the admin interface,
```bash
curl 'http://104.131.XX.XX:8086/query?u=root&p=root&q=select%20min(value)%20from%20a_series%20group%20by%20time(1d)&db=data'
```

So it seems that the JS client is not adding the `db=data` at the end of the URL for some reason? Any ideas why this may be? The client hasn't been changed in a while, and this code used to work just fine.. Also, I couldn't track down the Angular code that runs in the admin interface, but I guess it doesn't call the `influxdb.query()` method?

I'm using this version of the JS client: `http://get.influxdb.org/influxdb-latest.js`

**Comment by otoolep** [comment]
When you say "used to work", do you mean against 0.8? The 0.9 API changed significantly and it is possible this client code will not work with 0.9.

**Comment by bbirand** [comment]
I was using the CDN version of the JS library: `http://get.influxdb.org/influxdb-latest.js`. My 0.9-rc20 version came with an Angular app that references a version named `influxdb-0.8.999.js`. 

So the solution to my problem of the `db` field not getting appended to the URL was to use the `queryDatabase` call instead of `query`. This method exists in `influxdb-0.8.999.js` but not in `influxdb-latest.js`. I also searched the `influxdb-js` repo for this method, but can't find it there either:

https://github.com/influxdb/influxdb-js/search?q=queryDatabase

I still don't know what's going on. Isn't the "unstable" version of the JS code that comes with the influxd RC binary not tracked on github? (Also not documented...). And the `influxdb-latest.js` does not work with the 0.9 version of the server (i.e., it's the latest stable release?).

Thanks!

**Comment by otoolep** [comment]
Packages marked "latest" are only for 0.8 software, since 0.9 is considered
unstable and is in alpha stage right now. So using that JS client with
InfluxDB 0.9 will not work.

Philip

**Comment by beckettsean** [comment]
Not a bug with `influxdb/influxdb`. Please open an issue on the client repo.

**Comment by bbirand** [comment]
Thanks for the reply, Philip! I gathered as much about the JS versions. 
The "latest" one somehow did work with my RC version a few weeks back, 
but then was changed (which is completely legit - I should have used a 
local copy).

The mystery for me is now to get the source code of 
"influxdb-0.8.999.js" which, despite it's name, is actually for v0.9. 
This version doesn't seem to be on any of the branches of the 
influxdb-js repo?

As a general note, having a version be a "release candidate" is very 
different to calling it "alpha". Given that the documentation is 
non-existent, I suppose it's closer to that than to an RC release:
http://influxdb.com/docs/v0.9/clients/javascript.html

I would not have downloaded this version if it was appropriately labeled 
"alpha" so as not to get into these types of issues.


otoolep wrote:
